web/src/test/java/com/epam/rft/atsy/web/messageresolution/MessageKeyResolverImplTest.java

```diff
@@ -92,4 +94,24 @@ public void resolveMessageShouldResolveButNotInterpolateMessageWhenMessageKeyExi
     // Then
     assertThat(message, equalTo(EXAMPLE_WITH_NAME_MESSAGE));
   }
+
+  @Test
+  public void resolveMessageOrDefaultShouldReturnMessageWhenMessageKeyExists() {
+    // When
+    String message =
+        messageKeyResolver.resolveMessageOrDefault(EXAMPLE_MESSAGE_KEY, DEFAULT_MESSAGE);
+
+    // Then
+    assertThat(message, equalTo(EXAMPLE_MESSAGE));
+  }
+
+  @Test
+  public void resolveMessageOrDefaultShouldReturnMessageKeyWhenMessageResolutionFails() {
+    // When
+    String message =
+        messageKeyResolver.resolveMessageOrDefault(NON_EXISTENT_MESSAGE_KEY, DEFAULT_MESSAGE);
+
+    // Then
+    assertThat(message, equalTo("[ ! " + NON_EXISTENT_MESSAGE_KEY + " ! ]"));
```

Comment:
The [ ! and ! ] delimiters could be public constants in the message key resolver implementation and you could use them in tests.